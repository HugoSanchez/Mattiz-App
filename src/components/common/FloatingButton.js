import React from 'react';
import { View, TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/SimpleLineIcons';

import colors from '../../constants/colors'

// Props:
// @ onPress:  - Required. Function to execute.
// @ iconName: - Required. Icon to render. 

const FloatingButton = props => {
    return (
        <View>
            <TouchableOpacity
                activeOpacity={0.7}
                onPress={ props.onPress }
                style={styles.buttonStyle}>
                    <Icon 
                        name={ props.iconName } 
                        size={25} 
                        color={colors.palleteLightGreen} 
                        style={styles.iconStyle} /> 
            </TouchableOpacity>
        </View>
    );
}

const styles = {
    buttonStyle: {
        position: 'absolute',
        alignItems: 'center',
        justifyContent: 'center',
        width: 50,
        height: 50,
        borderRadius: 25,
        right: 25,
        bottom: 25,
        backgroundColor: colors.palleteDarkGreen,
        shadowColor: '#000',
        shadowOffset: { width: 5, height: 5 },
        shadowOpacity: 0.29,
        shadowRadius: 2.65,
        elevation: 2,
      },
      iconStyle: {
        marginTop: 3,
        marginRight: 3
    }
};

export default FloatingButton;