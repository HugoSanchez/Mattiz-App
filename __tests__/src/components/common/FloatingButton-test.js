import 'react-native'
import React from 'react'
import FloatingButton from '../../../../src/components/common/FloatingButton'
import {render, fireEvent} from '@testing-library/react-native'

test('Renders with title', () => {
	const {container} = render(<FloatingButton />)
	expect(container.children[0]).toMatchSnapshot()
})
