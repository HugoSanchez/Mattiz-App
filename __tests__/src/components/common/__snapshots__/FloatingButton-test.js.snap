// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders with title 1`] = `
<View>
  <TouchableOpacity
    activeOpacity={0.7}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#154053",
        "borderRadius": 25,
        "bottom": 25,
        "elevation": 2,
        "height": 50,
        "justifyContent": "center",
        "position": "absolute",
        "right": 25,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 5,
          "width": 5,
        },
        "shadowOpacity": 0.29,
        "shadowRadius": 2.65,
        "width": 50,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#D9F7EE",
            "fontSize": 25,
          },
          Object {
            "marginRight": 3,
            "marginTop": 3,
          },
          Object {
            "fontFamily": "simple-line-icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </TouchableOpacity>
</View>
`;
